### ISSUE TITLE
Failed to lex - Unterminated string, expected a '"' in my additional class

### ISSUE THREAD

**Issue by szelected** [jheruty/hscpp#15]
I've got this error message during hot swap, and I don't know what to do with it.
The standard compilation with VS2019 going well, only when I save an opengl widget that uses this additional class...
please advise !

**Comment by jheruty** [comment]
Is this issue coming from your compiler? It appears to be the same as an error from the hscpp lexer:
https://github.com/jheruty/hscpp/blob/7850c3acb2a694422fb2fc06559c6819ef64f758/src/preprocessor/LangError.cpp#L12

Do you have preprocessing enabled?

Would it be possible to take a look at the code? Perhaps there are cases that produce an unterminated string that are valid C++, which I did not account for.

**Comment by jheruty** [comment]
I see that my current code does not handle multiline strings:
```
"such as\
this one"
```
I will add this case, and also consider making certain preprocessor errors non-fatal.

**Comment by szelected** [comment]
Thank you for your reply.
Well, I think I did sorted out. Definitely it's my fault.
I write a function called GetExePath
used like this -> std::string exePath = this->GetExePath() + "\\";
The double backslash between quotes caused the issue and now it's working with single quotes '\\'

to be more specific:

MusicPlayer::MusicPlayer() {
    std::string exePath = this->GetExePath() + '\\';
    std::string mp3filename = exePath + "BODNB_01.mp3";
    bool initaudio = olc::SOUND::InitialiseAudio(44100, 2, 16, 1024);
    sample_id = olc::SOUND::LoadMp3FileSample(mp3filename);
    std::cout << "Music loaded..." << std::endl;
}

....

the get exepath goes like this :

std::string MusicPlayer::GetExePath() {
    TCHAR szExeFileName[MAX_PATH];
    GetModuleFileName(NULL, szExeFileName, MAX_PATH);
    std::string s = szExeFileName;
    return s.substr(0, s.find_last_of("\\/"));
    //return (std::string)"hello!";
}

seems to be whenever an std::string contains  "\\" the hotswap stopping to work.

to replicate:
simple make an std::string s = "\\"; - anywhere in your code.
note that my character set is : Use Multi-Byte Character Set

Please let me know your thoughts.

**Comment by jheruty** [comment]
That's a great find, and is a bug in the Lexer code. What was going on is that I did not handle escaped backslashes, so the Lexer interpreted `"\\"` as a slash and then an escaped quote.

I have added the case where the backslash is escaped.

Future improvements for this issue could be:
1) Better error message, telling you where the unterminated string began.
2) Reducing Lexing to only be done within hscpp macros and for `#include` statements.
